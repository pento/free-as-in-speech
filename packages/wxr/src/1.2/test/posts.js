/**
 * External dependencies
 */
const FDBFactory = require( 'fake-indexeddb/lib/FDBFactory' );

/**
 * Internal dependencies
 */
const { WXRDriver } = require( '../index.js' );

describe( 'Posts', () => {
	beforeEach( () => {
		window.indexedDB = new FDBFactory();
	} );

	test( 'All fields are written as expected', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.addCategory( {
			slug: 'some-cat',
			name: 'Some Category',
		} );

		wxr.addTag( {
			slug: 'some-tag',
			name: 'Some Tag',
		} );

		wxr.addTerm( {
			taxonomy: 'my-tax',
			slug: 'some-term',
			name: 'Some Term',
		} );

		wxr.addPost( {
			id: 1,
			title: 'A Post Title',
			link: 'https://wordpress.org/',
			date: '1970-01-01T00:00:01.000Z',
			author: 'pento',
			guid: 'should-be-a-proper-guid',
			description: 'This will not be written', // Non-writeable field.
			content: 'The post content',
			excerpt: 'A brief excerpt from the content',
			postDate: '1970-01-01T00:00:02.000Z',
			date_gmt: '1970-01-01T00:00:03.000Z',
			modified: '1970-01-01T00:00:04.000Z',
			modified_gmt: '1970-01-01T00:00:05.000Z',
			comment_status: 'closed',
			ping_status: 'open',
			status: 'draft',
			menu_order: 1,
			type: 'post',
			password: 'a-strong-password',
			sticky: true,
			attachment_url: 'https://make.wordpress.org', // Won't be written for this post type.
			terms: [
				{
					type: 'category',
					slug: 'some-cat',
					name: 'Some Category',
				},
				{
					type: 'tag',
					slug: 'some-tag',
					name: 'Some Tag',
				},
				{
					type: 'my-tax',
					slug: 'some-term',
					name: 'Some Term',
				},
			],
			meta: [ { key: 'talk about', value: 'pop music' } ],
		} );

		const xml = await wxr.export();

		expect( xml ).toMatchSnapshot();
	} );
} );
