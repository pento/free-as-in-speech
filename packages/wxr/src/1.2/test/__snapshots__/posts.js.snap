// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts All fields are written as expected 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<wp:category>
		<wp:category_nicename><![CDATA[some-cat]]></wp:category_nicename>
		<wp:category_parent><![CDATA[]]></wp:category_parent>
		<wp:cat_name><![CDATA[Some Category]]></wp:cat_name>
	</wp:category>
	<wp:tag>
		<wp:tag_slug><![CDATA[some-tag]]></wp:tag_slug>
		<wp:tag_name><![CDATA[Some Tag]]></wp:tag_name>
	</wp:tag>
	<wp:term>
		<wp:tag_taxonomy><![CDATA[my-tax]]></wp:tag_taxonomy>
		<wp:term_slug><![CDATA[some-term]]></wp:term_slug>
		<wp:term_parent><![CDATA[]]></wp:term_parent>
		<wp:term_name><![CDATA[Some Term]]></wp:term_name>
	</wp:term>
	<item>
		<wp:post_id>1</wp:post_id>
		<title><![CDATA[A Post Title]]></title>
		<link><![CDATA[https://wordpress.org/]]></link>
		<pubDate><![CDATA[Thu, 01 Jan 1970 00:00:01 GMT]]></pubDate>
		<dc:creator><![CDATA[pento]]></dc:creator>
		<guid isPermaLink=\\"false\\"><![CDATA[should-be-a-proper-guid]]></guid>
		<description></description>
		<content:encoded><![CDATA[The post content]]></content:encoded>
		<excerpt:encoded><![CDATA[A brief excerpt from the content]]></excerpt:encoded>
		<wp:post_date><![CDATA[1970-01-01 00:00:02]]></wp:post_date>
		<wp:post_date_gmt><![CDATA[1970-01-01 00:00:03]]></wp:post_date_gmt>
		<wp:post_modified><![CDATA[1970-01-01 00:00:04]]></wp:post_modified>
		<wp:post_modified_gmt><![CDATA[1970-01-01 00:00:05]]></wp:post_modified_gmt>
		<wp:comment_status><![CDATA[closed]]></wp:comment_status>
		<wp:ping_status><![CDATA[open]]></wp:ping_status>
		<wp:status><![CDATA[draft]]></wp:status>
		<wp:menu_order>1</wp:menu_order>
		<wp:post_type><![CDATA[post]]></wp:post_type>
		<wp:post_password><![CDATA[a-strong-password]]></wp:post_password>
		<wp:is_sticky>NaN</wp:is_sticky>
		<wp:attachment_url><![CDATA[https://make.wordpress.org]]></wp:attachment_url>
		<category domain=\\"category\\" nicename=\\"some-cat\\">Some Category</category>
		<category domain=\\"tag\\" nicename=\\"some-tag\\">Some Tag</category>
		<category domain=\\"my-tax\\" nicename=\\"some-term\\">Some Term</category>
		<wp:postmeta>
			<wp:meta_key>talk about</wp:meta_key>
			<wp:meta_value>pop music</wp:meta_value>
		</wp:postmeta>
	</item>
</channel>
</rss>"
`;
